### DISCUSSION TITLE
How to optimize performance for repeated rendering on live chat messages?

### DISCUSSION THREAD

**Discussion by bradleybernard** [gonzalezreal/swift-markdown-ui#261] [Q&A]
I'm using this library to render Markdown for my new iOS app that is an AI assistant. The bot messages that come in, come in as a `String` that includes markdown text. 

Each new chunk of the response triggers a SwiftUI view body update and therefore triggers the whole Markdown parsing + rendering process.

Is there any way to optimize for markdown content parsing & rendering, knowing I'm going to be building a tree over time?

1. Every 10ms new `String` `contents` come in
2. Pass to my SwiftUI View
3. SwiftUI View renders `Markdown(contents)`
4. `Markdown` SwiftUI view parses and renders `MarkdownContent`
5. Repeat 50-200 times above ^

Could I resume parsing where I left off? Sharing parsing artifacts/memory between subsequent parsing calls? Same with rendering? Caching previous results with some logic there. Curious what other folks are doing.

**Comment by yousefkhater** [comment]
Hi @bradleybernard  did you find a solution for this ?
